### ISSUE TITLE
Cannot apply config using fortimanager_securityconsole_install_package

### ISSUE THREAD

**Issue by kostas-rf** [fortinetdev/terraform-provider-fortimanager#57]
Hi Team,

We are deploying our policies to Fortimanager using Terraform for a while now. Recently, we upgraded to the latest provider version (v1.12.0) to fix another provider bug. Since then it seems we can no longer use the `fortimanager_securityconsole_install_package` resource to apply the config updates to the Fortigate firewalls. Please find the details below:

Terraform resource:
```
resource "fortimanager_securityconsole_install_package" "install_package" {
  depends_on = [
    fortimanager_packages_firewall_policy.policy
  ]

  fmgadom  = "root"
  flags    = ["none"]
  pkg      = var.install_pkg_name

  scope {
    name = var.install_pkg_name
    vdom = "root"
  }

  force_recreate = uuid()
}
```

Terraform plan:
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # fortimanager_securityconsole_install_package.install_package will be created
  + resource "fortimanager_securityconsole_install_package" "install_package" {
      + dynamic_sort_subtable = "false"
      + flags                 = [
          + "none",
        ]
      + fmgadom               = "root"
      + force_recreate        = (known after apply)
      + id                    = (known after apply)
      + pkg                   = "nonprd-gates"

      + scope {
          + name = "nonprd-gates"
          + vdom = "root"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

Error:
```
fortimanager_securityconsole_install_package.install_package: Creating...
2024-06-24T14:37:20.650Z [INFO]  Starting apply for fortimanager_securityconsole_install_package.install_package
2024-06-24T14:37:20.658Z [DEBUG] fortimanager_securityconsole_install_package.install_package: applying the planned Create change
2024-06-24T14:37:20.658Z [TRACE] GRPCProvider: ApplyResourceChange
2024-06-24T14:37:20.661Z [TRACE] provider.terraform-provider-fortimanager_v1.12.0: Received request: tf_provider_addr=provider tf_req_id=07ee6937-1305-553b-eb84-53680b05881a tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov5/tf5server/server.go:833 @module=sdk.proto tf_proto_version=5.4 tf_resource_type=fortimanager_securityconsole_install_package timestamp=2024-06-24T14:37:20.659Z
2024-06-24T14:37:20.662Z [TRACE] provider.terraform-provider-fortimanager_v1.12.0: Sending request downstream: tf_provider_addr=provider tf_req_id=07ee6937-1305-553b-eb84-53680b05881a tf_proto_version=5.4 tf_resource_type=fortimanager_securityconsole_install_package tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:22 @module=sdk.proto timestamp=2024-06-24T14:37:20.659Z
2024-06-24T14:37:20.662Z [TRACE] provider.terraform-provider-fortimanager_v1.12.0: Calling downstream: tf_resource_type=fortimanager_securityconsole_install_package tf_provider_addr=provider tf_req_id=07ee6937-1305-553b-eb84-53680b05881a tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:908 @module=sdk.helper_schema timestamp=2024-06-24T14:37:20.660Z
2024-06-24T14:37:20.662Z [INFO]  provider.terraform-provider-fortimanager_v1.12.0: 2024/06/24 14:37:20 [INFO] Request infomation: /securityconsole/install/package:    &map[adom:[root] flags:[none] pkg:[nonprd-gates] scope:[map[name:nonprd-gates vdom:root]]]: timestamp=2024-06-24T14:37:20.660Z
2024-06-24T14:37:20.662Z [INFO]  provider.terraform-provider-fortimanager_v1.12.0: 2024/06/24 14:37:20 [INFO] Request URL: /securityconsole/install/package: timestamp=2024-06-24T14:37:20.660Z
2024-06-24T14:37:20.699Z [TRACE] provider.terraform-provider-fortimanager_v1.12.0: Called downstream: tf_provider_addr=provider @module=sdk.helper_schema tf_req_id=07ee6937-1305-553b-eb84-53680b05881a tf_resource_type=fortimanager_securityconsole_install_package tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:910 timestamp=2024-06-24T14:37:20.698Z
2024-06-24T14:37:20.699Z [TRACE] provider.terraform-provider-fortimanager_v1.12.0: Received downstream response: tf_resource_type=fortimanager_securityconsole_install_package tf_rpc=ApplyResourceChange @module=sdk.proto diagnostic_error_count=1 tf_req_id=07ee6937-1305-553b-eb84-53680b05881a tf_req_duration_ms=38 @caller=github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:42 diagnostic_warning_count=0 tf_proto_version=5.4 tf_provider_addr=provider timestamp=2024-06-24T14:37:20.698Z
2024-06-24T14:37:20.699Z [ERROR] provider.terraform-provider-fortimanager_v1.12.0: Response contains error diagnostic: diagnostic_detail= tf_provider_addr=provider tf_req_id=07ee6937-1305-553b-eb84-53680b05881a tf_resource_type=fortimanager_securityconsole_install_package tf_rpc=ApplyResourceChange @module=sdk.proto diagnostic_severity=ERROR diagnostic_summary="Error updating SecurityconsoleInstallPackage resource:
err -10: The data is invalid for selected url" tf_proto_version=5.4 @caller=github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov5/internal/diag/diagnostics.go:58 timestamp=2024-06-24T14:37:20.698Z
2024-06-24T14:37:20.699Z [TRACE] maybeTainted: fortimanager_securityconsole_install_package.install_package encountered an error during creation, so it is now marked as tainted
2024-06-24T14:37:20.699Z [TRACE] provider.terraform-provider-fortimanager_v1.12.0: Served request: tf_req_id=07ee6937-1305-553b-eb84-53680b05881a tf_resource_type=fortimanager_securityconsole_install_package @module=sdk.proto tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov5/tf5server/server.go:859 tf_proto_version=5.4 tf_provider_addr=provider timestamp=2024-06-24T14:37:20.698Z
2024-06-24T14:37:20.699Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for fortimanager_securityconsole_install_package.install_package
2024-06-24T14:37:20.699Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: removing state object for fortimanager_securityconsole_install_package.install_package
2024-06-24T14:37:20.699Z [TRACE] evalApplyProvisioners: fortimanager_securityconsole_install_package.install_package is tainted, so skipping provisioning
2024-06-24T14:37:20.699Z [TRACE] maybeTainted: fortimanager_securityconsole_install_package.install_package was already tainted, so nothing to do
2024-06-24T14:37:20.700Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for fortimanager_securityconsole_install_package.install_package
2024-06-24T14:37:20.700Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: removing state object for fortimanager_securityconsole_install_package.install_package
2024-06-24T14:37:20.700Z [ERROR] vertex "fortimanager_securityconsole_install_package.install_package" error: Error updating SecurityconsoleInstallPackage resource:
err -10: The data is invalid for selected url
2024-06-24T14:37:20.700Z [TRACE] vertex "fortimanager_securityconsole_install_package.install_package": visit complete, with errors
2024-06-24T14:37:20.700Z [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/fortinetdev/fortimanager\"] (close)" errored, so skipping
2024-06-24T14:37:20.700Z [TRACE] dag/walk: upstream of "root" errored, so skipping
╷
│ Error: Error updating SecurityconsoleInstallPackage resource:
│ err -10: The data is invalid for selected url
│
│   with fortimanager_securityconsole_install_package.install_package,
│   on install_package.tf line 1, in resource "fortimanager_securityconsole_install_package" "install_package":
│    1: resource "fortimanager_securityconsole_install_package" "install_package" {
│
╵
```

This used to work in previous versions of the provider with no issues, although i cannot go back to the previous version now due to other bugs.

Of course, when going back the Fortimanager and installing the package manually works as expected.

Provider version: 1.12.0
Terraform version: 1.12
Fortimanager version: v7.2.5

Thanks!

**Comment by MaxxLiu22** [comment]
Hi @kostas-rf ,

Thank you for bringing this issue to our attention, and we sincerely apologize for any inconvenience caused. The issue has arisen due to a data type change in the arguments ``adom`` and ``pkg`` during the evolution of FortiManager. Our development team is aware of this change and is actively working on improvements.

In the meantime, we kindly request you to use the [FortiManager JSON Generic API](https://registry.terraform.io/providers/fortinetdev/fortimanager/latest/docs/resources/fortimanager_json_generic_api) as a temporary solution.

Thank you for your understanding and patience.

```
resource "fortimanager_json_generic_api" "tfname" {
  json_content = <<JSON
{
    "method": "exec",
    "params": [
        {
            "data": {
                "adom": "root",
                "pkg": "${var.install_pkg_name}",
                "scope": [
                    {
                        "name": "${var.install_pkg_name}",
                        "vdom": "root2"
                    }
                ]
            },
            "url": "/securityconsole/install/package"
        }
    ]
}
JSON
}
```

**Comment by kostas-rf** [comment]
Hi @MaxxLiu22,

Thank you for your suggestion, this fixed the immediate issue.
Do you have an ETA on a permanent fix?

**Comment by MaxxLiu22** [comment]
Hi @kostas-rf ,

We recently released a minor version update, Terraform FMG 1.12.1, which should address this issue. Please let us know if you continue to experience any problems. Sorry for any inconvenience.

Thanks,
Maxx

**Comment by MaxxLiu22** [comment]
Hi @kostas-rf 

I will proceed with closing this case. However, please feel free to reopen it or open a new case if the issue persists or if you have any further questions.

Thanks,
Maxx
